resource/environment/integration.js

```diff
@@ -241,6 +245,141 @@ function upgrade( slack, envelope ) {
 	} );
 }
 
+function getTemplate( token, catalogOwner, catalog, info ) {
```

Comment:
there are a lot of nested functions here. maybe consider putting this in a module?